packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/ExtractIconFromDataTests.m

```diff
@@ -0,0 +1,175 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import google_maps_flutter_ios;
+@import google_maps_flutter_ios.Test;
+@import XCTest;
+#import <OCMock/OCMock.h>
+
+@interface ExtractIconFromDataTests : XCTestCase
+- (UIImage *)createOnePixelImage;
+- (NSData *)convertImageToPNGData:(UIImage *)image;
+@end
+
+@implementation ExtractIconFromDataTests
```

Comment:
is there a test to verify the resizing vs scaling threshold logic?

Reply:
I changed from calculating the difference between aspect ratios using epsilon to check the original size against target size with one pixel tolerance, to have better results.
Also added test cases to test this check.